### PR TITLE
Work3

### PR DESCRIPTION
**Pull request by ArtemKa2208**
homework 3

### CODE REVIEW

File: destructuring.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+const sarah = {
+      name: 'Sarah',
+      job: {
+        name: 'Andersen',
+        hiringDate: '23.07.2020'
+      }
+}
+    
+function printJobName(user) {
+      let {
```
Comment (StanislawVictorovich) [inline destructuring.js]: don't use additional variables, do it directly with input param

File: destructuring.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+const sarah = {
+      name: 'Sarah',
+      job: {
+        name: 'Andersen',
+        hiringDate: '23.07.2020'
+      }
+}
+    
+function printJobName({name,job:{name: job_name,hiringDate},}) {
```
Comment (StanislawVictorovich) [inline destructuring.js]: It's correct way but remove unused props

File: mapSet.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+const arr = [{id: 1},{id: 2},{id: 3}, {id: 1}];
+
+function removeDuplicates(arr){
```
Comment (StanislawVictorovich) [inline mapSet.js:3]: function must return Array instead of Map

File: destructuring.js (hunk 1 above)
Comment (ArtemKa2208) [reply destructuring.js]: done

File: destructuring.js (hunk 2 above)
Comment (ArtemKa2208) [reply destructuring.js]: done

File: mapSet.js (hunk 3 above)
Comment (ArtemKa2208) [reply mapSet.js:3]: done

File: mapSet.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+      let map = new Map();
+      let newArr = [];
+      for(let i = 0; i < arr.length; i++){
```
Comment (StanislawVictorovich) [inline mapSet.js:6]: you have 2 cycles - this is not optimized

Comment (ArtemKa2208) [reply mapSet.js:6]: done

**Pull request merged**